internal/env/parser_test.go

```diff
@@ -0,0 +1,311 @@
+package env
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+
+	"github.com/mainflux/mainflux/internal/clients/grpc"
+	"github.com/mainflux/mainflux/internal/server"
+	"github.com/stretchr/testify/assert"
+)
+
+var errNotDuration error = errors.New("unable to parse duration")
+
+func TestParseServerConfig(t *testing.T) {
+	tests := []struct {
+		description    string
+		config         *server.Config
+		expectedConfig *server.Config
+		options        []Options
+		err            error
+	}{
+		{
+			"Parsing with Server Config",
+			&server.Config{},
+			&server.Config{
+				Host:     "localhost",
+				Port:     "8080",
+				CertFile: "cert",
+				KeyFile:  "key",
+			},
+			[]Options{
+				{
+					Environment: map[string]string{
+						"HOST":        "localhost",
+						"PORT":        "8080",
+						"SERVER_CERT": "cert",
+						"SERVER_KEY":  "key",
+					},
+				},
+			},
+			nil,
+		},
+		{
+			"Parsing with Server Config with Prefix",
+			&server.Config{},
+			&server.Config{
+				Host:     "localhost",
+				Port:     "8080",
+				CertFile: "cert",
+				KeyFile:  "key",
+			},
+			[]Options{
+				{
+					Environment: map[string]string{
+						"MF-HOST":        "localhost",
+						"MF-PORT":        "8080",
+						"MF-SERVER_CERT": "cert",
+						"MF-SERVER_KEY":  "key",
+					},
+					Prefix: "MF-",
+				},
+			},
+			nil,
+		},
+		{
+			"Parsing with Server Config with Alt-Prefix",
+			&server.Config{},
+			&server.Config{
+				Host:     "localhost",
+				Port:     "8080",
+				CertFile: "cert",
+				KeyFile:  "key",
+			},
+			[]Options{
+				{
+					Environment: map[string]string{
+						"MF-HOST":        "localhost",
+						"MF-HTTP-PORT":   "8080",
+						"MF-SERVER_CERT": "cert",
+						"MF-SERVER_KEY":  "key",
+					},
+					Prefix:    "MF-",
+					AltPrefix: "MF-HTTP-",
+				},
+			},
+			nil,
+		},
+		{
+			"Parsing with conflicting configs",
+			&server.Config{},
+			&server.Config{
+				Host:     "localhost",
+				Port:     "8080",
+				CertFile: "cert",
+				KeyFile:  "key",
+			},
+			[]Options{
+				{
+					Environment: map[string]string{
+						"PORT":                "",
+						"MF-PORT":             "",
+						"MF-HOST":             "localhost",
+						"MF-HTTP-PORT":        "8080",
+						"MF-SERVER_CERT":      "",
+						"MF-SERVER_KEY":       "",
+						"MF-HTTP-SERVER_CERT": "cert",
+						"MF-HTTP-SERVER_KEY":  "key",
+					},
+					Prefix:    "MF-",
+					AltPrefix: "MF-HTTP-",
+				},
+			},
+			nil,
+		},
+	}
+	for _, test := range tests {
+		err := Parse(test.config, test.options...)
+		if test.err != nil {
+			assert.Error(t, err, "expected error but got nil")
+		} else {
+			assert.NoError(t, err, fmt.Sprintf("expected no error but got %v", err))
+		}
```

Comment:
```suggestion
assert.Equal(t, tc.err, err, fmt.Sprintf("expected error %s, but got %s", tc.err, err))
```

Reply:
The errors generated are concatenated, they are also not exported from the package so I can't use errors.Contains(). I could copy the entire  error and assert equality, but this current assertion still achieves similar results

Reply:
Makes sense, wrapping errors would not bring benefits here. Please use `switch` instead of `if/else`, just to be aligned with the rest of the code.